### PR TITLE
Feature/1919 create generic caching in selfservice portal

### PR DESCRIPTION
**Pull request by kralle333**
closes dfds/cloudplatform/issues/1919

## Additional Review Notes
Besides having a solution for dealing with adding new sources this PR also contains:

- Very simple implementation of displaying total aws resources count column in MyCapabilities section
- Added some functions to help display aws resource info: `getAwsResourcesTotalCountForCapability`, `getAwsResourceCountsForCapability`, `getAwsResourceCountForCapability`
- Now calling renamed endpoint: `my-capabilities-costs` instead of `my-capability-costs`

depends on dfds/selfservice-api#51

### CODE REVIEW

File: src/src/MetricsWrapper.js [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+  getAwsResourceCountsForCapability(capabilityId) {
+    const metric = this.metrics.get(MetricsWrapper.ResourceCountsKey);
+    const counts = metric.capabilitiesMap.get(capabilityId);
+    if (counts === undefined) {
+      return new Map();
+    }
+    return counts;
+  }
+
+  getAwsResourceCountForCapability(capabilityId, resourceId) {
```
Comment (andreasfrisch) [inline src/src/MetricsWrapper.js:142]: Can we have multiple counts of the same resourceId?
I may misunderstand what is meant by resourceId in this case

**Review by andreasfrisch — CHANGES_REQUESTED** [review]
Minor suggestion for code improvement.
Otherwise nice 🎉

File: src/src/AppContext.js [hunk 2]
```diff
@@ -79,19 +80,34 @@ function AppProvider({ children }) {
...
+    metricsWrapper.tryUpdateMetrics().then(() => {
+      setAppStatus((prev) => ({
+        ...prev,
+        ...{
+          hasLoadedMyCapabilitiesCosts: metricsWrapper.hasLoaded(
+            MetricsWrapper.CostsKey,
+          ),
+        },
+      }));
+      setAppStatus((prev) => ({
```
[the remark this answers is not in the document]
Comment (kralle333) [reply src/src/AppContext.js]: I have cleaned this up ;)

File: src/src/MetricsWrapper.js [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+          .set(resourceCount.resourceId, resourceCount.resourceCount);
+      });
+    });
+    if (resourceCountsMap.size > 0) {
+      metric.updateCapabilityMap(now, resourceCountsMap);
+      this.metrics.set(MetricsWrapper.ResourceCountsKey, metric);
+    }
+  }
+
+  async #tryUpdateMyCapabilityCosts(now) {
```
[the remark this answers is not in the document]
Comment (kralle333) [reply src/src/MetricsWrapper.js]: I tried refactoring this, what do you think?

File: src/src/MetricsWrapper.js (hunk 1 above)
Comment (kralle333) [reply src/src/MetricsWrapper.js:135]: Here is an example of resource ids and counts:
image

A real example for using this function could be: "I would like to get the count of databases for this capability". I believe that corresponding Id is: `aws_rds_db_total`

**Review by andreasfrisch — APPROVED** [review]
Sweet 👍

**Pull request merged**